src/TAMSEquationSystem.C

```diff
@@ -0,0 +1,615 @@
+//-------- initial_work ----------------------------------------------------
+//--------------------------------------------------------------------------
+void
+TAMSEquationSystem::initialize_mdot()
+{}
+/*{
+  // Don't do this if it's a restart and average_mdot has been defined...
+  if (resetAverages_) {
+
+    stk::mesh::MetaData& meta_data = realm_.meta_data();
+    if (realm_.realmUsesEdges_) {
+      ScalarFieldType* massFlowRate_ = meta_data.get_field<ScalarFieldType>(
+        stk::topology::EDGE_RANK, "mass_flow_rate");
+
+      stk::mesh::Selector s_all_nodes =
+        (meta_data.locally_owned_part() | meta_data.globally_shared_part()) &
+        stk::mesh::selectField(*avgMdot_);
+
+      stk::mesh::BucketVector const& edge_buckets =
+        realm_.get_buckets(stk::topology::EDGE_RANK, s_all_nodes);
+      for (stk::mesh::BucketVector::const_iterator ib = edge_buckets.begin();
+           ib != edge_buckets.end(); ++ib) {
+        stk::mesh::Bucket& b = **ib;
+        const stk::mesh::Bucket::size_type length = b.size();
+
+        const double* mdot = stk::mesh::field_data(*massFlowRate_, b);
+        double* avgMdot = stk::mesh::field_data(*avgMdot_, b);
+
+        for (stk::mesh::Bucket::size_type k = 0; k < length; ++k)
+          avgMdot[k] = mdot[k];
+      }
+    } else {
+      GenericFieldType* massFlowRateScs_ = meta_data.get_field<GenericFieldType>(
+        stk::topology::ELEMENT_RANK, "mass_flow_rate_scs");
+
+      // define some common selectors
+      stk::mesh::Selector s_all_elem =
+        (meta_data.locally_owned_part() | meta_data.globally_shared_part()) &
+        stk::mesh::selectField(*avgMdotScs_);
+
+      stk::mesh::BucketVector const& elem_buckets =
+        realm_.get_buckets(stk::topology::ELEMENT_RANK, s_all_elem);
+      for (stk::mesh::BucketVector::const_iterator ib = elem_buckets.begin();
+           ib != elem_buckets.end(); ++ib) {
+        stk::mesh::Bucket& b = **ib;
+        const stk::mesh::Bucket::size_type length = b.size();
+
+        // extract master element
+        MasterElement* meSCS =
+          sierra::nalu::MasterElementRepo::get_surface_master_element(
+            b.topology());
+
+        // extract master element specifics
+        const int numScsIp = meSCS->num_integration_points();
+
+        for (stk::mesh::Bucket::size_type k = 0; k < length; ++k) {
+          double* avgMdotScs = stk::mesh::field_data(*avgMdotScs_, b, k);
+          const double* mdotScs = stk::mesh::field_data(*massFlowRateScs_, b, k);
+
+          for (int ip = 0; ip < numScsIp; ip++)
+            avgMdotScs[ip] = mdotScs[ip];
+        }
+      }
+    }
+  }
+}*/
+
+//--------------------------------------------------------------------------
+//-------- compute_metric_tensor() -----------------------------------------
+//--------------------------------------------------------------------------
+void
+TAMSEquationSystem::compute_metric_tensor()
+{
+  if (NULL != metricTensorAlgDriver_)
```

Comment:
`metricTensorAlgDriver_` is created in the initializer list of the constructor, is there a situation where it would be `nullptr`?

Reply:
nope. removed the if condition for this one and the others